ddl_checker/main.go

```diff
@@ -0,0 +1,214 @@
+
+	host     = flag.String("host", "127.0.0.1", "MySQL host")
+	port     = flag.Int("port", 3306, "MySQL port")
+	username = flag.String("user", "root", "User name")
+	password = flag.String("password", "", "Password")
+	schema   = flag.String("schema", "", "Schema")
+)
+
+const (
+	WelcomeInfo = "ExecutableChecker: Check if SQL can be successfully executed by TiDB\n" +
+		"Copyright 2018 PingCAP, Inc.\n\n" +
+		"You can switch modes using the `SETMOD` command.\n" +
+		"Auto mode: The program will automatically synchronize the dependent table structure from MYSQL " +
+		"and delete the conflict table\n" +
+		"Prompt mode: The program will ask you before synchronizing the dependent table structure from MYSQL\n" +
+		"Manual mode: This program does not perform anything other than executing the input SQL.\n\n" +
+		"SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).\n\n"
+)
+
+func main() {
+	initialise()
+	printWelcome()
+	mainLoop()
+	destroy()
+}
+
+func initialise() {
+	flag.Parse()
+	var err error
+	reader = bufio.NewReader(os.Stdin)
+	executableChecker, err = ddl_checker.NewExecutableChecker()
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't create ExecutableChecker: %s\n", err.Error())
+		os.Exit(1)
+	}
+	executableChecker.Execute("use test;")
+	db, err = sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		*username, *password, *host, *port, *schema))
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't open mysql database: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+}
+
+func destroy() {
+	executableChecker.Close()
+	db.Close()
+}
+
+func mainLoop() {
+	var input string
+	var err error
+	for cont := true; cont; cont = handler(input) {
+		fmt.Printf("[%s] > ", modeName())
+		input, err = reader.ReadString(';')
+		if err != nil {
+			fmt.Printf("[DDLChecker] Read stdin error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}
+}
+func handler(input string) bool {
+	// cmd exit
+	lowerTrimInput := strings.ToLower(strings.TrimFunc(input, func(r rune) bool {
+		return unicode.IsSpace(r) || r == ';'
+	}))
+	if lowerTrimInput == "exit" {
+		return false
+	}
+	// cmd setmod
+	if strings.HasPrefix(lowerTrimInput, "setmod") {
+		switch strings.TrimSpace(lowerTrimInput[6:]) {
+		case "auto":
+			modCode = 0
+		case "prompt":
+			modCode = 1
+		case "manual":
+			modCode = 2
+		default:
+			fmt.Println("SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).")
+		}
+		return true
+	}
+	if modCode != 2 {
+		// auto and query mod
+		stmt, err := executableChecker.Parse(input)
+		if err != nil {
+			fmt.Printf("[DDLChecker] SQL parse error: %s\n", err.Error())
+			return true
+		}
+		neededTable := ddl_checker.GetTableNeededExist(stmt)
+		nonNeededTable := ddl_checker.GetTableNeededNonExist(stmt)
+		if modCode == 0 || (modCode == 1 && queryAutoSync(neededTable, nonNeededTable)) {
+			syncTablesFromMysql(neededTable)
```

Comment:
if some errors happen in these two function `syncTablesFromMysql` and `dropTables`, would it continue to execute?

Reply:
yes,it will print error info,and continue to try executing the input sql.

Reply:
it wonldn't